lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py

```diff
@@ -0,0 +1,285 @@
+            origin_origin_access_identity_ids = [oai.get('Id') for oai in
+                                                 origin_access_identities]
+            for origin_access_identity_id in origin_origin_access_identity_ids:
+                oai_config = (
+                    self.__cloudfront_facts_mgr.get_origin_access_identity_config(
+                        origin_access_identity_id))
+                temp_caller_reference = oai_config.get(
+                    'CloudFrontOriginAccessIdentityConfig').get(
+                    'CallerReference')
+                if temp_caller_reference == caller_reference:
+                    return origin_access_identity_id
+        except Exception as e:
+            self.module.fail_json(
+                msg="error getting origin access identity from " +
+                "caller_reference - " + str(e) + "\n" + traceback.format_exc())
+
+    def validate_comment(self, comment):
+        if comment is None:
+            comment = (
+                "origin access identity created by Ansible with datetime " +
+                datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f'))
+        return comment
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+
+    argument_spec.update(dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        origin_access_identity_id=dict(required=False, default=None,
+                                       type='str'),
+        caller_reference=dict(required=False, default=None, type='str'),
+        comment=dict(required=False, default=None, type='str')
+    ))
+
+    result = {}
+    e_tag = None
+    changed = False
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=False)
+    service_mgr = CloudFrontOriginAccessIdentityServiceManager(module)
+    validation_mgr = CloudFrontOriginAccessIdentityValidationManager(module)
+
+    state = module.params.get('state')
+    caller_reference = module.params.get('caller_reference')
+    comment = module.params.get('comment')
+    origin_access_identity_id = module.params.get('origin_access_identity_id')
+
+    if origin_access_identity_id is None and caller_reference is not None:
+        origin_access_identity_id = (
+            validation_mgr.validate_origin_access_identity_id_from_caller_reference(
+                caller_reference))
+
+    e_tag = validation_mgr.validate_etag_from_origin_access_identity_id(
+        origin_access_identity_id)
+    comment = validation_mgr.validate_comment(comment)
+
+    if state == 'present':
+        if origin_access_identity_id is not None and e_tag is not None:
+            result = service_mgr.update_origin_access_identity(
+                caller_reference, comment, origin_access_identity_id, e_tag)
+        else:
+            result = service_mgr.create_origin_access_identity(
+                caller_reference, comment)
+        changed = True
```

Comment:
It looks like this module is not idempotent. Running the same task twice with `state: present` or `state: absent` will show changed every time.